pkg/cfn/builder/vpc.go

```diff
@@ -108,27 +112,25 @@ func (c *ClusterResourceSet) addResourcesForVPC() {
 
 	c.addSubnets(refPublicRT, api.SubnetTopologyPublic, c.spec.VPC.Subnets.Public)
 
-	c.newResource("NATIP", &gfn.AWSEC2EIP{
-		Domain: gfn.NewString("vpc"),
-	})
-	refNG := c.newResource("NATGateway", &gfn.AWSEC2NatGateway{
-		AllocationId: gfn.MakeFnGetAttString("NATIP.AllocationId"),
-		// A multi-AZ NAT Gateway is possible, but it's not very
-		// clear from the docs how to achieve it
-		SubnetId: c.subnets[api.SubnetTopologyPublic][0],
-	})
+	c.addNATGateways()
 
-	refPrivateRT := c.newResource("PrivateRouteTable", &gfn.AWSEC2RouteTable{
-		VpcId: c.vpc,
-	})
+	c.addSubnets(nil, api.SubnetTopologyPrivate, c.spec.VPC.Subnets.Private)
+}
 
-	c.newResource("PrivateSubnetRoute", &gfn.AWSEC2Route{
-		RouteTableId:         refPrivateRT,
-		DestinationCidrBlock: internetCIDR,
-		NatGatewayId:         refNG,
-	})
+func (c *ClusterResourceSet) addNATGateways() {
+
+	switch c.spec.VPC.NAT.Gateway {
+
+	case api.NATHighlyAvailable:
+		c.haNAT()
+	case api.NATSingle:
+		c.singleNAT()
+	case api.NATDisable:
+		c.noNAT()
+	default:
+		logger.Critical("%s is not a valid NAT gateway mode", c.spec.VPC.NAT.Gateway)
```

Comment:
This should just return an error :)

Reply:
Sorry, just to be clear, something like this ? 

```
func (c *ClusterResourceSet) addNATGateways() error {

	switch *c.spec.VPC.NAT.Gateway {

	case api.ClusterHighlyAvailableNAT:
		c.haNAT()
	case api.ClusterSingleNAT:
		c.singleNAT()
	case api.ClusterDisableNAT:
		c.noNAT()
	default:
		return errors.New(*c.spec.VPC.NAT.Gateway + "is not a valid NAT gateway mode")
	}
	return nil
}
```

Reply:
👍